File: src/main/java/org/activecraft/activecraftcore/listener/CommandListener.java
package org.activecraft.activecraftcore.listener;

import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.messagesv2.ActiveCraftMessage;
import org.activecraft.activecraftcore.messagesv2.PlayerMessageFormatter;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.messagesv2.ActiveCraftMessage;
import org.activecraft.activecraftcore.messagesv2.PlayerMessageFormatter;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;

public class CommandListener implements Listener {

    @EventHandler(priority = EventPriority.MONITOR)
    public void onCommandSent(PlayerCommandPreprocessEvent event) {
        Player executingPlayer = event.getPlayer();
        Profilev2 executingPlayerProfile = Profilev2.of(executingPlayer);
        String eventMessage = event.getMessage();
        ActiveCraftMessage accoreMessage = ActiveCraftCore.getInstance().getActiveCraftMessagev2();

        for (Player player : Bukkit.getOnlinePlayers()) {
            Profilev2 profile = Profilev2.of(player);
            if (!profile.canReceiveLog()) continue;
            if (!player.hasPermission("activecraft.log")) continue;
            player.sendMessage(
                    executingPlayerProfile.getMessageSupplier(accoreMessage).getFormatted(
                            "command.log.format",
                            new PlayerMessageFormatter(accoreMessage, executingPlayerProfile).addFormatterPattern("command", eventMessage)
                    ));
        }
    }

}

File: src/main/java/org/activecraft/activecraftcore/listener/CommandListener.kt
package org.activecraft.activecraftcore.listener

import org.activecraft.activecraftcore.ActiveCraftCore
import org.activecraft.activecraftcore.messages.PlayerMessageFormatter
import org.activecraft.activecraftcore.playermanagement.Profile.Companion.of
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerCommandPreprocessEvent

class CommandListener : Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    fun onCommandSent(event: PlayerCommandPreprocessEvent) {
        val executingPlayer = event.player
        val executingPlayerProfile = of(executingPlayer)
        val eventMessage = event.message
        val accoreMessage = ActiveCraftCore.INSTANCE.activeCraftMessage
        for (player in Bukkit.getOnlinePlayers()) {
            val profile = of(player)
            if (!profile.receiveLog) continue
            if (!player.hasPermission("activecraft.log")) continue
            player.sendMessage(
                executingPlayerProfile.getMessageSupplier(accoreMessage!!).getFormatted(
                    "command.log.format",
                    PlayerMessageFormatter(accoreMessage, executingPlayerProfile).addFormatterPattern(
                        "command",
                        eventMessage
                    )
                )
            )
        }
    }
}